cmake/Options.cmake

```diff
@@ -10,12 +10,18 @@ else()
   set(LINUX OFF)
 endif()
 
+include(CMakeDependentOption)
+
+# TODO: Default to ON if CUDA available.
+option(TP_ENABLE_CUDA "Enable support for CUDA tensors" OFF)
```

Comment:
Should we try to use a different "style" for the option name than what we use for transports/channels, to avoid confusion? (I would also like to change the ones for transports/channels, to add a `_TRANSPORT` and `_CHANNEL` suffix, but that's another matter).

PyTorch controls its CUDA build with the `USE_CUDA` env var, so we could do the same and call it `TP_USE_CUDA`. Or `TP_BUILD_WITH_CUDA`. Or `TP_SUPPORT_CUDA`.

The one thing that will certainly change when this flag is enabled is that the pipe will support the CudaTensor type, right? (The other thing that will change is that we will try to build the CUDA channels, but the user could override that). So maybe we should make it explicit and call the flag `TP_ENABLE_CUDA_TENSORS`? Or `TP_SUPPORT_CUDA_TENSORS`?

Reply:
Whichever is fine to me, except `TP_{ENABLE,SUPPORT}_CUDA_TENSORS`, as it might be a bit too narrowing. I don't have a preference between `USE`/`SUPPORT`/`ENABLE`.

Reply:
Let's say `TP_USE_CUDA` for this one? To align with PyTorch